#167750040 Setup Eslint for Linting and Babel for Transpiling code

sa-ma opened the pull request:
#### Description of Tasks to be completed?
- Setup Eslint for linting
- Setup Babel for transpiling es5 to es6

 #### How should this be manually tested?
- Clone repo
- cd into the cloned directory
- Write es6 code and follow coding conventions to test babel and Eslint
- Run `npm run start:dev`

 #### Any background context you want to provide?
None
 #### What are the relevant pivotal tracker stories?
 [#167750040](https://www.pivotaltracker.com/story/show/167750040)
#### Screenshots (if appropriate)
None
 #### Questions:
None

sa-ma committed:
chore(eslint): Setup Eslint and Babel

Setup eslint to lint files
Setup babel to transpile es5 to es6

[Finishes #167750040]

eokwukwe reviewed (changes requested):
Good job so far @sa-ma. Please refer back to the `wiki` on commit messages and confirm whether you're supposed to capitalize any word in your commit.

Also, please rename that `server` folder to what it was before `src`. That was not the name change we talked about. What we said we're going to change is the name of the server entry file from `index.js` to `server.js`.

sa-ma commented:
Should I create another commit or amend the commit message?

eokwukwe commented on .eslintrc:
```diff
@@ -0,0 +1,39 @@
+{
+  "root": true,
+  "extends": "airbnb-base",
+  "env": {
```
dmfm

sa-ma committed:
chore(folderName): undo folder name change

- rename server folder to src
- rename app.js to index.js
- update package.json to reflect the name change

[Fixes #167750040]

Oliver-ke commented on src/index.js:
```diff
@@ -1,93 +1,18 @@
-        }
-    });
-});
+app.get('/', (req, res) => res.json({ status: res.statusCode, message: 'Welcome to Wolfsbane' }));
```
Nice work bro, I think you should have set the status code directly as in
res.status(200).json({ status: 200, message: 'welcome to wolfsbane'})
doing this is more explicit, allowing us to know the actual status code while sending the response

eokwukwe replied:
@Oliver-ke, for now, we can leave it the way it is. We'll need to improve the welcome page later.

oyewoas reviewed (commented):
nice work @sa-ma

feobaby reviewed (dismissed):
Good job **Samaila**, I can see you installed dependencies for eslint and babel and I also looked out for two things; the config files for **eslint** and **babel** which are present. Also, your PR write-up meets the standard format. 
_Please watch out for commit messages better to avoid further mistakes._

Oliver-ke reviewed (commented):
Nice work Sa-ma, Pull request tested, More eslint rule should be added later as the project grows

uthdev commented on .babelrc:
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["@babel/preset-env"]
```
Nice work @sa-ma. I can see you are using babel 7. Won't we be needing @babel/transform-runtime and @babel/runtime? To avoid regenerator runtime error when transpiling async functions. since @eokwukwe said we'll be working with async await.

eokwukwe replied:
@sa-ma. You need to implement @uthdev feedback. Do that so that I can merge this `PR`

sa-ma replied:
Fixed @uthdev  I added a target version for the babel-presets so it will generate all the plugins needed. There won't be any need to install the plugins individually

uthdev reviewed (commented):
What do you think? Cause I think I've come up against such error when working on the ADC.

sa-ma committed:
chore(babel): edit babel config

- edit babel config to support async/await

[Fixes #167750040]

uthdev reviewed (dismissed):
Nice one @sa-ma. I think this is good enough to move forward.

eokwukwe reviewed (changes requested):
@sa-ma you also need to return these part of the code that was removed as we'll need their functionality for the codebase:

```
const isProduction = process.env.NODE_ENV === "production";
```

```
/// catch 404 and forward to error handler
app.use((req, res, next) => {
    const err = new Error("Not Found");
    err.status = 404;
    next(err);
});
```

```
// development error handler
// will print stacktrace
if (!isProduction) {
    app.use(function(err, req, res, next) {
        console.log(err.stack);

        res.status(err.status || 500);

        res.json({
            errors: {
                message: err.message,
                error: err
            }
        });
    });
}
```

```
// production error handler
// no stacktraces leaked to user
app.use(function(err, req, res, next) {
    res.status(err.status || 500);
    res.json({
        errors: {
            message: err.message,
            error: {}
        }
    });
});
```

You'll need to add `NODE_ENV=development` in the script that is used to run the app in development mode. So it will look like this:
```
NODE_ENV=development nodemon --exec babel-node ./src/index.js
```

**Remember to change the _function_ definition to `ES6` arrow function**

darasimiolaifa reviewed (dismissed):
Looks good to me Sama. Good job. Well done.

sa-ma committed:
chore(devEnvironment): setup error handler

- setup error handler to catch errors on endpoints and print
appropriate messagess based on the environment

[Fixes  #167750040]

eokwukwe commented on .babelrc:
```diff
@@ -0,0 +1,12 @@
+{
+  "presets": [
+    [
+      "@babel/preset-env",
```
kekekdk

oyewoas reviewed (dismissed):
nice one

uthdev reviewed (dismissed):
Nice job. @sama. @eokwukwe, please check out this PR.

feobaby reviewed (dismissed):
Great job Sama!

eokwukwe reviewed (changes requested):
@sa-ma please remove those dependencies that we're not going to use from `package.json`.

darasimiolaifa reviewed (dismissed):
Alright Bro. Good to go. Well done.

eokwukwe commented on package.json:
```diff
@@ -1,30 +1,34 @@
-    "passport-local": "^1.0.0",
-    "request": "^2.87.0",
-    "underscore": "^1.9.1"
+    "passport-local": "^1.0.0"
```
We'll not need `passport-local`

eokwukwe commented on package.json:
```diff
@@ -1,30 +1,34 @@
     "cors": "^2.8.4",
     "dotenv": "^6.0.0",
-    "ejs": "^2.6.1",
     "errorhandler": "^1.5.0",
```
We'll not need `errorhandler`
